scripts/west_commands/runners/bossac.py

```diff
@@ -51,14 +54,27 @@ def do_run(self, command, **kwargs):
             raise NotImplementedError(msg)
 
         self.require(self.bossac)
-
         if platform.system() == 'Linux':
             self.require('stty')
             cmd_stty = ['stty', '-F', self.port, 'raw', 'ispeed', '1200',
                         'ospeed', '1200', 'cs8', '-cstopb', 'ignpar', 'eol', '255',
                         'eof', '255']
             self.check_call(cmd_stty)
 
+            # Wait until target reboots
+            sleep(0.250)
+            # Wait until port re-enumerates
+            retries = 0
+            while (retries < 3):
+                try:
+                    self.check_call(cmd_stty)
```

Comment:
Am I imagining something, or did this used to just check if the file existed previously? Why is it calling stty again?

Reply:
It was either there for checking if the file existed or for putting those ardurino-compatible boards in upgrade mode for flashing (though the latter might be only a coincidence). If the call to stty does indeed reboot a board into upgrade mode it causes it to briefly disappear from /dev/.  I included the following calls of stty in order to make sure that it re-appears by the time we try to move on to calling bossac.

Reply:
I don't know what this stty line is doing, but is it safe to call multiple times like this? It seems like just checking `os.path.isfile(self.port)` might work. Is there a reason to call stty instead of doing that?

Asking mostly out of concern about side effects. Any education you can provide me would be very welcome. Thank you.